torch/csrc/jit/codegen/cuda/type.cpp

```diff
@@ -87,6 +87,9 @@ ValType promote_type(const ValType& t1, const ValType& t2) {
       (t1 == ValType::Scalar || t1 == ValType::NamedScalar)) {
     return ValType::Scalar;
   }
+  if (t1 == ValType::NamedScalar && t2 == ValType::NamedScalar) {
```

Comment:
Looks like an important fix 

I'm guessing we probably don't want it in this PR though.

Reply:
The change makes sense to me. I haven't seen an exception here before, so looks like there has just been no such case so far.

Reply:
The exception showed up in the following PR to this one. I'm too lazy to split it out into a separate PR line so leaving it in here.